File: (Java)DynamicBroadcastReceiver/app/src/main/java/com/dilip/javadynamicbroadcastreceiver/MainActivity.java
package com.dilip.javadynamicbroadcastreceiver;

import android.content.Intent;
import android.content.IntentFilter;
import android.net.ConnectivityManager;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    MyBroadcastReceiver myBroadcastReceiver = new MyBroadcastReceiver();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        IntentFilter intentFilter = new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION);
        intentFilter.addAction(Intent.ACTION_TIME_TICK);

        registerReceiver(myBroadcastReceiver, intentFilter);

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unregisterReceiver(myBroadcastReceiver);
    }

    public void sendBroadcast(View view) {

    }
}

File: DynamicBroadcastReceiver/app/src/main/java/com/dilip/dynamicbroadcastreceiver/MainActivity.kt
package com.dilip.dynamicbroadcastreceiver

import android.content.Intent
import android.content.IntentFilter
import android.net.ConnectivityManager
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.dilip.dynamicbroadcastreceiver.databinding.ActivityMainBinding


class MainActivity : AppCompatActivity() {
    private var exampleReceiver: ExampleReceiver? = null
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)

        exampleReceiver = ExampleReceiver()

        // Create an IntentFilter to specify the broadcasts we're interested in
        val filter = IntentFilter()
        filter.addAction(ConnectivityManager.CONNECTIVITY_ACTION)
        filter.addAction(Intent.ACTION_TIME_TICK)

        // Register the receiver with the system
        registerReceiver(exampleReceiver, filter)
    }

    override fun onDestroy() {
        super.onDestroy()

        // Unregister the receiver when the activity is destroyed
        unregisterReceiver(exampleReceiver)
    }
}
